Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py

```diff
@@ -563,9 +563,21 @@ def projects(self):
 
         return result
 
+    def valid_keywords(self):
+        response = self.session.get(
+            '{}/rest/field/bug/{}{}'.format(self.url, 'keywords', self._login_arguments(required=False)),
+            timeout=self.timeout,
+        )
+        if response.status_code // 100 == 4 and self._logins_left:
+            self._logins_left -= 1
+        if response.status_code // 100 != 2:
+            sys.stderr.write("Failed to retrieve keywords list'\n")
+            return []
+        return [value.get('name', '') for value in response.json().get('fields')[0].get('values', [])]
+
     def create(
         self, title, description,
-        project=None, component=None, version=None, assign=True,
+        project=None, component=None, version=None, assign=True, keywords=[]
```

Comment:
Weird nit: you shouldn't assign an empty list in an argument like this in Python. It creates by value/by reference confusions because doing `keywords.add` in the function will end up persisting to the next function call. Your current code doesn't have this problem, but it's good to be defensive. Should do this pattern that you see a few different places in our codebase instead:
```
def ...(self, ..., keywords=None):
    keywords = keywords or []
    ...
```

Reply:
Oo good to know for next time. Thanks!